crypto/engine/eng_pkey.c

```diff
@@ -90,7 +91,8 @@ EVP_PKEY *ENGINE_load_public_key(ENGINE *e, const char *key_id,
         ERR_raise(ERR_LIB_ENGINE, ERR_R_PASSED_NULL_PARAMETER);
         return 0;
     }
-    CRYPTO_THREAD_write_lock(global_engine_lock);
+    if (!CRYPTO_THREAD_write_lock(global_engine_lock))
+        return 0;
```

Comment:
NULL

Reply:
changed all 'return 0' in this function to 'return NULL'